aws/resource_aws_route_table_test.go

```diff
@@ -270,6 +270,40 @@ func TestAccAWSRouteTable_panicEmptyRoute(t *testing.T) {
 	})
 }
 
+func TestAccAWSRouteTable_Route_ConfigMode(t *testing.T) {
+	var routeTable1, routeTable2, routeTable3 ec2.RouteTable
+	resourceName := "aws_route_table.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckRouteTableDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSRouteTableConfigRouteConfigModeBlocks(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRouteTableExists(resourceName, &routeTable1),
+					resource.TestCheckResourceAttr(resourceName, "route.#", "2"),
+				),
+			},
+			{
+				Config: testAccAWSRouteTableConfigRouteConfigModeNoBlocks(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRouteTableExists(resourceName, &routeTable2),
+					resource.TestCheckResourceAttr(resourceName, "route.#", "2"),
```

Comment:
Why is the count two here if there are no route attributes in the config?

Reply:
This attribute is marked `Computed: true`: https://github.com/terraform-providers/terraform-provider-aws/blob/c5c8b5c69fd40fca3d4b9dfec6d2e546dc1783cb/aws/resource_aws_route_table.go#L46

For better or worse, the historical reason for this is because we support two mutually exclusive configuration styles for EC2 Route Table Routes:
* Inline `route` configuration blocks, each managing a Route, within the `aws_route_table` resource
* Individual Route managing [`aws_route` resources](https://www.terraform.io/docs/providers/aws/r/route.html), usually paired with modularized Terraform configuration

In the case of the latter, the `Computed: true` is there to prevent the need for configuring `ignore_changes`, e.g.

```hcl
# Example required configuration if Computed: true was not set for route attribute
resource "aws_route_table" "example" {
  # ... other configuration, except any route configuration blocks ...

  lifecycle {
    ignore_changes = "route"
  }
}

resource "aws_route" "example" {
  # ... other configuration ...

  route_table_id = "${aws_route_table.example.id}"
}
```

Reply:
Got it! I was looking at it as a separate test :man_facepalming:

 All good here and thanks for the backgound on the two resources `aws_route` and the inline route in `aws_route_table`. That was not at first obvious in reading the documentation for `aws_route_table`.